pkg/controller/certificates/sync.go

```diff
@@ -877,7 +877,11 @@ func (c *certificateRequestManager) setSecretValues(ctx context.Context, crt *cm
 
 	s.Data[corev1.TLSPrivateKeyKey] = data.pk
 	s.Data[corev1.TLSCertKey] = data.cert
-	s.Data[cmmeta.TLSCAKey] = data.ca
+	if data.ca != nil && len(data.ca) > 0 {
```

Comment:
We don't need to check for nil, instead we can just check if the length > 0 (as nil slices have a length of 0 already) 😄

Reply:
Good to know